[SecurityInsights] Generated XML cref omits Models qualifier for DataConnectorsCheckRequirements

### Description

The generated SecurityInsights ArmClient extension XML documentation omits the `Models.` namespace qualifier for `DataConnectorsCheckRequirements` in `cref` values.

In `sdk/securityinsights/Azure.ResourceManager.SecurityInsights/src/Generated/Extensions/SecurityInsightsExtensions.cs`, the public method signatures use `Models.DataConnectorsCheckRequirements`, but the mocking `cref` entries use unqualified `DataConnectorsCheckRequirements`:

```csharp
/// <see cref="MockableSecurityInsightsArmClient.PostAsync(ResourceIdentifier, DataConnectorsCheckRequirements, CancellationToken)"/>
public static async Task<Response<DataConnectorRequirementsState>> PostAsync(
    this ArmClient client,
    ResourceIdentifier scope,
    Models.DataConnectorsCheckRequirements dataConnectorsCheckRequirements,
    CancellationToken cancellationToken = default)
```

There is also an internal root-namespace type named `DataConnectorsCheckRequirements` generated under `Generated/RestOperations`, so the unqualified cref cannot resolve to the public model method signature.

### Error

```text
Generated/Extensions/SecurityInsightsExtensions.cs(3346,71): error CS1574: XML comment has cref attribute 'PostAsync(ResourceIdentifier, DataConnectorsCheckRequirements, CancellationToken)' that could not be resolved
```

The same issue exists for the sync `Post(...)` cref.

### Expected behavior

The generated XML `cref` should include the model namespace qualifier, matching the generated method signature:

```csharp
MockableSecurityInsightsArmClient.PostAsync(ResourceIdentifier, Models.DataConnectorsCheckRequirements, CancellationToken)
MockableSecurityInsightsArmClient.Post(ResourceIdentifier, Models.DataConnectorsCheckRequirements, CancellationToken)
```

### Repro context

Observed while regenerating `Azure.ResourceManager.SecurityInsights` in the `securityinsights-mpg-migration` branch using the local management generator workflow.

Regen command used locally:

```powershell
pwsh -NoProfile -File eng/packages/http-client-csharp-mgmt/eng/scripts/RegenSdkLocal.ps1 -Services "Azure.ResourceManager.SecurityInsights" -LocalSpecRepoPath C:\Users\pashao\GIT\azure-rest-api-specs -SaveInputs
```

This appears to be a generator XML documentation qualification issue rather than a SecurityInsights handwritten customization issue.

--- a/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/CSharpTypeExtensions.cs
+++ b/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/CSharpTypeExtensions.cs
@@ -63,12 +63,24 @@ namespace Azure.Generator.Management.Utilities
 
         public static string GetXmlDocTypeName(this CSharpType type)
         {
+            var typeName = GetTypeName(type);
+
             // For nullable value types, we need to append '?' to the type name for XML documentation
             if (type.IsValueType && type.IsNullable)
             {
-                return $"{type.Name}?";
+                return $"{typeName}?";
             }
-            return type.Name;
+            return typeName;
+        }
+
+        private static string GetTypeName(CSharpType type)
+        {
+            if (string.IsNullOrEmpty(type.Namespace))
+            {
+                return type.Name;
+            }
+
+            return $"{type.Namespace}.{type.Name}";
         }
     }
 }

--- a/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/test/Providers/ExtensionProviderTests.cs
+++ b/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/test/Providers/ExtensionProviderTests.cs
@@ -4,6 +4,7 @@
 using Azure.Generator.Management.Providers;
 using Azure.Generator.Management.Tests.Common;
 using Azure.Generator.Management.Tests.TestHelpers;
+using Azure.Generator.Management.Utilities;
 using Humanizer;
 using NUnit.Framework;
 
@@ -88,6 +89,17 @@ namespace Azure.Generator.Management.Tests.Providers
                 Assert.That(mockableResource.Methods.Count, Is.GreaterThan(0),
                     $"MockableResourceProvider '{mockableResource.Name}' should have at least one method to be included in the output.");
             }
+        }
+
+        [TestCase]
+        public void Verify_MockingCrefQualifiesModelParameters()
+        {
+            var model = InputFactory.Model("ResponseType", clientNamespace: "Samples.Models");
+            var plugin = ManagementMockHelpers.LoadMockPlugin(inputModels: () => [model]);
+            var modelType = plugin.Object.TypeFactory.CreateCSharpType(model);
+            Assert.That(modelType, Is.Not.Null);
+
+            Assert.That(modelType!.GetXmlDocTypeName(), Is.EqualTo("Samples.Models.ResponseType"));
         }
     }
 }

